woocommerce/woocommerce-gutenberg-products-block
# [Experimental] Mini Cart Icon Control via `theme.json`.

This is an experimental PR that allows control over the Mini Cart Icon via a theme's `theme.json` file. 

### Test Plan

- Apply PR and run `npm start`.
- In the current theme, add the following config in the `settings` property in the `theme.json` file:
  ```json
  "blocks": {
  	"woocommerce/mini-cart": {
  		"custom": {
  			"icon": "miniCartBasket"
  		}
  	}
  }
  ```
- Confirm that the icon for the Mini Cart block has updated to use a basket icon.
- Adjust the `icon` setting from `miniCartBasket` to `miniCartBag` and confirm that the Mini Cart block now uses a bag icon.
- Confirm no JS or PHP errors.

**Note:** This current implementation also updates the icon for the block in the inserter. We likely want to keep this icon the same, as it can potentially lead to confusion for what a user is looking for when adding a block in the editor. I've left it in here for the sake of experimentation, however.

### To Do

- [ ] Confirm it is ok to add configs that we’re not using in CSS in this way.
- [ ] Investigate if it is possible to set special settings in the JSON that will be ignored.
- [ ] Confirm it is ok to include our settings on `settings/block/custom` or if it would be better to add a separate setting outside of  `settings`.
- [ ] Code cleanup (consider if and where we can put the PHP logic as well as the duplicate JS logic between the editor and front end).
- [ ] Explore the possibility of using a block filter to allow theme developers more options in extending/overriding the icon.
